src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -262,55 +250,51 @@ internal override ValueTask ConnectAsync(CancellationToken cancellationToken = d
                 _ => throw new Exception(SR.Format(SR.net_quic_unsupported_address_family, _remoteEndPoint.AddressFamily))
             };
 
-            QuicExceptionHelpers.ThrowIfFailed(
-                MsQuicApi.Api.ConnectionStartDelegate(
-                _ptr,
-                (ushort)af,
-                address,
-                (ushort)port),
-                "Failed to connect to peer.");
-
-            return new ValueTask(_connectTcs.Task);
-        }
-
-        private MsQuicStream StreamOpen(
-            QUIC_STREAM_OPEN_FLAG flags)
-        {
-            IntPtr streamPtr = IntPtr.Zero;
-            QuicExceptionHelpers.ThrowIfFailed(
-                MsQuicApi.Api.StreamOpenDelegate(
-                _ptr,
-                (uint)flags,
-                MsQuicStream.s_streamDelegate,
-                IntPtr.Zero,
-                out streamPtr),
-                "Failed to open stream to peer.");
-
-            return new MsQuicStream(this, flags, streamPtr, inbound: false);
-        }
-
-        private void SetCallbackHandler()
-        {
-            Debug.Assert(!_handle.IsAllocated, "callback handler allocated already");
-            _handle = GCHandle.Alloc(this);
+            _state.Connection = this;
+            try
+            {
+                uint status = MsQuicApi.Api.ConnectionStartDelegate(
```

Comment:
Would it make sense to have QuicStatus enum here? While the list of errors may grow, there are really only three interesting values IMHO. We do that in some other places and it make debugging much easier as you can see the enum instead of number.

Reply:
We already have something: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Quic/src/System/Net/Quic/Interop/MsQuicStatusCodes.cs
The codes differ on different platforms, that's why it's separate. Could we define an enum per platform and use it? I guess we could. Is it worth it? Depends on what we decide to do about the generated interop.
Otherwise, better messages are covered by https://github.com/dotnet/runtime/issues/32066.